src/sql/workbench/parts/dataExplorer/browser/connectionViewletPanel.ts

```diff
@@ -58,8 +56,10 @@ export class ConnectionViewletPanel extends ViewletPanel {
 			AddServerGroupAction.LABEL);
 		this._serverTreeView = this.instantiationService.createInstance(ServerTreeView);
```

Comment:
Shouldn't you only create this if there isn't already a tree view? So something like this :

```
if(!treeView) {
    this._serverTreeView = this.instantiationService.createInstance(ServerTreeView);
    this.objectExplorerService.registerServerTreeView(this._serverTreeView);
}
```

Reply:
(I'd also then do the getServerTreeView like this :

```
this._serverTreeView = this.objectExplorerService.getServerTreeView();
if(!this._serverTreeView) {
…
```

Reply:
The serverTreeView gets disposed when the connectionViewletPanel is created again so we do need to create it every time. We just need to not register it every time.

Reply:
But the registration is then to a disposed object at that point. Do we really need to dispose the tree view when its collapsed? Shouldn't we keep it around so we don't have to reinitialize everything every time they close/re-open?

---
In reply to: [276368154](https://github.com/Microsoft/azuredatastudio/pull/5076#discussion_r276368154) [](ancestors = 276368154)

Reply:
The serverTreeView is null every time ConnectionViewletPanel is made. The OEService still has the serverTreeView saved, however. The reason the ViewletPanel gets disposed because that's how the parent class `ViewContainerViewlet` works. And the panel gets disposed not when collapsed, but when removed from the Viewlet.

Reply:
Tested with mssql and postgres, works fine.